File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnMusicIdOptions.java

package com.gracenote.gnsdk;

/** 
** Configures options for {@link GnMusicId} 
*/ 
 
public class GnMusicIdOptions {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnMusicIdOptions(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnMusicIdOptions obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnMusicIdOptions(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

/** 
*  Indicate whether the MusicID query should be performed against local embedded databases or online. 
*  @param lookupMode  [in] One of the {@link GnLookupMode} values 
* 
*/ 
 
  public void lookupMode(GnLookupMode lookupMode) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_lookupMode__SWIG_0(swigCPtr, this, lookupMode.swigValue());
  }

/** 
*  Return current lookup mode for MusicID query 
* 
*/ 
 
  public GnLookupMode lookupMode() throws com.gracenote.gnsdk.GnException {
    return GnLookupMode.swigToEnum(gnsdk_javaJNI.GnMusicIdOptions_lookupMode__SWIG_1(swigCPtr, this));
  }

/** 
*  Set the lookup data value for the MusicID query. 
*  @param lookupData [in] One of the {@link GnLookupData} values 
*  @param bEnable    [in] Set lookup data 
* 
*/ 
 
  public void lookupData(GnLookupData lookupData, boolean bEnable) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_lookupData(swigCPtr, this, lookupData.swigValue(), bEnable);
  }

/** 
*  Set the preferred language for the returned results. 
*  @param preferredLanguage [in] One of the GNSDK language values 
* 
*/ 
 
  public void preferResultLanguage(GnLanguage preferredLanguage) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_preferResultLanguage(swigCPtr, this, preferredLanguage.swigValue());
  }

/** 
*  Set the preferred external ID of the returned results. 
*  Only available where single result is also requested. 
*  @param strExternalId [in] Gracenote external ID source name 
* 
*/ 
 
  public void preferResultExternalId(String strExternalId) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_preferResultExternalId(swigCPtr, this, strExternalId);
  }

/** 
*  Set that you would like cover art (if availble) in the returned results. 
*  @param bEnable [in] Set prefer cover art 
* 
*/ 
 
  public void preferResultCoverart(boolean bEnable) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_preferResultCoverart(swigCPtr, this, bEnable);
  }

/** 
*  Set that the response must return only a single best result. 
*  When enabled, a single full result is returned; when disabled, multiple partial results may be returned. 
*  @param bEnable [in] Set single result 
*  <p><b>Remarks:</b></p> 
*  If enabled, the MusicID library selects the single best result based on the query type and input. 
* 
*/ 
 
  public void resultSingle(boolean bEnable) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_resultSingle(swigCPtr, this, bEnable);
  }

/** 
*  Enable or disable revision check option. 
*  @param bEnable [in] Set revision check 
* <p> 
* Long Running Potential: Network I/O, File system I/O (for online query cache or local lookup) 
* 
*/ 
 
  public void revisionCheck(boolean bEnable) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_revisionCheck(swigCPtr, this, bEnable);
  }

/** 
*  Specfies whether a response must return a range of results beginning at a specified value. 
*  @param resultStart  [in] Result range start value 
*  <p><b>Remarks:</b></p> 
*  This option is useful for paging through results. 
*  <p><b>Note:</b></p> 
*  Gracenote Service enforces that the range start value must be less than or equal to the total 
*  number of results. If you specify a range start value that is greater than the total number of 
*  results, no results are returned. 
* 
*/ 
 
  public void resultRangeStart(long resultStart) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_resultRangeStart(swigCPtr, this, resultStart);
  }

/** 
*  Set the maximum number of results to return. 
*  @param resultCount  [in] Maximum results 
*  <p><b>Note:</b></p> 
*  Gracenote Service enforces its own maximum number of results that can vary over time and query types. 
*  This setting sets a maximum that can be as high as the Service maximum  but not higher. 
* 
*/ 
 
  public void resultCount(long resultCount) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_resultCount(swigCPtr, this, resultCount);
  }

/** 
* Set a specific network interface to use with this object's connections. This can be useful for 
* systems with multiple network interfaces. Otherwise, the operating system determines the interface to use. 
*  @param intfName [in] Local IP address or system name for the desired network interface 
* 
*/ 
 
  public void networkInterface(String intfName) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_networkInterface__SWIG_0(swigCPtr, this, intfName);
  }

/** 
* Return the network interface being used with this object's connections if one has been set. 
* If no specific network interface has been set, this option returns an empty string. 
* 
*/ 
 
  public String networkInterface() throws com.gracenote.gnsdk.GnException {
    return gnsdk_javaJNI.GnMusicIdOptions_networkInterface__SWIG_1(swigCPtr, this);
  }

/** 
* Specifies whether {@link GnMuisicId} processes received audio for music identification 
* suitability. 
* This option is experimental and subject to change 
* 
*/ 
 
  public void audioSuitabilityProcessing(boolean bEnable) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_audioSuitabilityProcessing(swigCPtr, this, bEnable);
  }

/** 
*  Set option using option name 
*  @param option   [in] Option name 
*  @param value	[in] Option value 
* 
*/ 
 
  public void custom(String option, String value) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_custom__SWIG_0(swigCPtr, this, option, value);
  }

/** 
*  Set option using option name 
*  @param option   [in] Option name 
*  @param bEnable	[in] Option enable true/false 
* 
*/ 
 
  public void custom(String option, boolean bEnable) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnMusicIdOptions_custom__SWIG_1(swigCPtr, this, option, bEnable);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnMusicIdOptions.cs

namespace GracenoteSDK {

/**
* \class GnMusicIdOptions
* Configures options for GnMusicId
*/
public class GnMusicIdOptions : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnMusicIdOptions(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnMusicIdOptions obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnMusicIdOptions() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnMusicIdOptions(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

/**
*  Indicate whether the MusicID query should be performed against local embedded databases or online.
*  @param lookupMode  [in] One of the #GnLookupMode values
*  @ingroup MusicId_Options
*/
  public void LookupMode(GnLookupMode lookupMode) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_LookupMode__SWIG_0(swigCPtr, (int)lookupMode);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Return current lookup mode for MusicID query
*  @ingroup MusicId_Options
*/
  public GnLookupMode LookupMode() {
    GnLookupMode ret = (GnLookupMode)gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_LookupMode__SWIG_1(swigCPtr);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
*  Set the lookup data value for the MusicID query.
*  @param lookupData [in] One of the #GnLookupData values
*  @param bEnable    [in] Set lookup data
*  @ingroup MusicId_Options
*/
  public void LookupData(GnLookupData lookupData, bool bEnable) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_LookupData(swigCPtr, (int)lookupData, bEnable);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Set the preferred language for the returned results.
*  @param preferredLanguage [in] One of the GNSDK language values
*  @ingroup MusicId_Options
*/
  public void PreferResultLanguage(GnLanguage preferredLanguage) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_PreferResultLanguage(swigCPtr, (int)preferredLanguage);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Set the preferred external ID of the returned results.
*  Only available where single result is also requested.
*  @param strExternalId [in] Gracenote external ID source name
*  @ingroup MusicId_Options
*/
  public void PreferResultExternalId(string strExternalId) {
  System.IntPtr tempstrExternalId = GnMarshalUTF8.NativeUtf8FromString(strExternalId);
    try {
      gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_PreferResultExternalId(swigCPtr, tempstrExternalId);
      if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    } finally {
 GnMarshalUTF8.ReleaseMarshaledUTF8String(tempstrExternalId);
    }
  }

/**
*  Set that you would like cover art (if availble) in the returned results.
*  @param bEnable [in] Set prefer cover art
*  @ingroup MusicId_Options
*/
  public void PreferResultCoverart(bool bEnable) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_PreferResultCoverart(swigCPtr, bEnable);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Set that the response must return only a single best result.
*  When enabled, a single full result is returned; when disabled, multiple partial results may be returned.
*  @param bEnable [in] Set single result
*  <p><b>Remarks:</b></p>
*  If enabled, the MusicID library selects the single best result based on the query type and input.
*  @ingroup MusicId_Options
*/
  public void ResultSingle(bool bEnable) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_ResultSingle(swigCPtr, bEnable);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Enable or disable revision check option.
*  @param bEnable [in] Set revision check
*
* Long Running Potential: Network I/O, File system I/O (for online query cache or local lookup)
*  @ingroup MusicId_Options
*/
  public void RevisionCheck(bool bEnable) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_RevisionCheck(swigCPtr, bEnable);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Specfies whether a response must return a range of results beginning at a specified value.
*  @param resultStart  [in] Result range start value
*  <p><b>Remarks:</b></p>
*  This option is useful for paging through results.
*  <p><b>Note:</b></p>
*  Gracenote Service enforces that the range start value must be less than or equal to the total
*  number of results. If you specify a range start value that is greater than the total number of
*  results, no results are returned.
*  @ingroup MusicId_Options
*/
  public void ResultRangeStart(uint resultStart) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_ResultRangeStart(swigCPtr, resultStart);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Set the maximum number of results to return.
*  @param resultCount  [in] Maximum results
*  <p><b>Note:</b></p>
*  Gracenote Service enforces its own maximum number of results that can vary over time and query types.
*  This setting sets a maximum that can be as high as the Service maximum  but not higher.
*  @ingroup MusicId_Options
*/
  public void ResultCount(uint resultCount) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_ResultCount(swigCPtr, resultCount);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Set a specific network interface to use with this object's connections. This can be useful for
* systems with multiple network interfaces. Otherwise, the operating system determines the interface to use.
*  @param intfName [in] Local IP address or system name for the desired network interface
*  @ingroup MusicId_Options
*/
  public void NetworkInterface(string intfName) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_NetworkInterface__SWIG_0(swigCPtr, intfName);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Return the network interface being used with this object's connections if one has been set.
* If no specific network interface has been set, this option returns an empty string.
*  @ingroup MusicId_Options
*/
  public string NetworkInterface() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_NetworkInterface__SWIG_1(swigCPtr);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
* Specifies whether GnMuisicId processes received audio for music identification
* suitability.
* This option is experimental and subject to change
* @ingroup MusicId_Options
*/
  public void AudioSuitabilityProcessing(bool bEnable) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_AudioSuitabilityProcessing(swigCPtr, bEnable);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Set option using option name
*  @param option   [in] Option name
*  @param value	[in] Option value
*  @ingroup MusicId_Options
*/
  public void Custom(string option, string value) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_Custom__SWIG_0(swigCPtr, option, value);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Set option using option name
*  @param option   [in] Option name
*  @param bEnable	[in] Option enable true/false
*  @ingroup MusicId_Options
*/
  public void Custom(string option, bool bEnable) {
    gnsdk_csharp_marshalPINVOKE.GnMusicIdOptions_Custom__SWIG_1(swigCPtr, option, bEnable);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

}

}
